### PR TITLE
BarChart: Use new data error view component to show actions in panel edit

### PR DESCRIPTION
**Pull request by torkelo**
Screenshot from 2021-11-29 18-19-15

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

File: public/app/features/panel/components/PanelDataErrorView.tsx [hunk 1]
```diff
@@ -68,6 +68,10 @@ function getMessageFor(
     return 'No data';
   }
 
+  if (needsStringField && !dataSummary.hasStringField) {
+    return 'Data is missing a string field';
```
Comment (torkelo) [inline public/app/features/panel/components/PanelDataErrorView.tsx:72]: wonder if there is a better way to phrase this that is not too wordy. "Visualization needs a string field and none could be found in current query results"  is better but too wordy

Comment (axelavargas) [reply public/app/features/panel/components/PanelDataErrorView.tsx:72]: 🤔 I have seen another data error when data is missing a time field:

> Data does not have a time field

Maybe we can follow the same here?

> Data does not have a string field

But tbh I am not sure is better 😅 , as you said the wordy sentence is better.

**Review by axelavargas — APPROVED** [review]
